akka-stream/src/main/scala/akka/stream/impl/Timers.scala

```diff
@@ -25,10 +25,19 @@ import scala.concurrent.duration.{ Duration, FiniteDuration }
  *  - otherwise, these streams do not interfere with the element flow, ordinary completion or failure
  */
 @InternalApi private[akka] object Timers {
-  private def idleTimeoutCheckInterval(timeout: FiniteDuration): FiniteDuration = {
+
+  /**
+   * Given a timeout computes how often the check should be run without causing
+   * excessive load or loosing timeout precision.
+   */
+  private[akka] def timeoutCheckInterval(timeout: FiniteDuration): FiniteDuration = {
     import scala.concurrent.duration._
     FiniteDuration(
-      math.min(math.max(timeout.toNanos / 8, 100.millis.toNanos), timeout.toNanos / 2),
+      math.min(
+        math.min(
+          math.max(timeout.toNanos / 8, 100.millis.toNanos),
+          timeout.toNanos / 2),
+        1.second.toNanos),
```

Comment:
I think doing it with an
```
 if (1 second or more) 1 second
 else min-max-calculation
```

Would be a little bit easier to read.

Reply:
(not in general, but here where there is also some calculations going on)

Reply:
👍